packages/studio/src/components/HighlightingContainer.tsx

```diff
@@ -87,6 +104,20 @@ class HighlightingClass extends Component<HighlightingProps> {
   }
 
   componentDidUpdate(): void {
+    const childNode = getDOMNode(this);
+    if (!childNode) {
+      return;
+    }
+    const rect = rectToJson(childNode.getBoundingClientRect());
```

Comment:
can some of this logic be consolidated with the `highlightSelf` logic? it looks pretty similar except for whether the active component logic

Reply:
I think the name `highlightSelf` is a little misleading for what that function currently does. To me, it seems to highlight itself and set itself to the active component. I was thinking, we could remove the active component check and setting and possibly move that into a separate function. Then the click handler would call the modified `highlightSelf` and perform the active component checking/setting. And `componentDidUpdate` would call `highlightSelf` if the component is a selected component and it's rect has changed. We could also move the `isRectUpdated` check into `highlightSelf`

Reply:
I like that structure, just updated it.